src/test/ccl/ut_cclut_mock.inc

```diff
@@ -419,6 +455,78 @@ subroutine test_cclutExecuteProgramWithMocks_mock_subroutine(null)
         cclut::internalRecord->item, 1)
 end ;test_cclutExecuteProgramWithMocks_mock_subroutine
 
+/* test_cclutExecuteProgramWithMocks_same_table_column_name *************************************************
+*  Scenario: Replaces only table names when the table and column share the same name                        *
+************************************************************************************************************/
+subroutine test_cclutExecuteProgramWithMocks_same_table_column_name(null)
+    record cclut_testRecord (
+        1 table_name = vc
+        1 value = vc
+    ) with protect
+    call cclutDefineMockTable("sample_table", "sample_table", "vc")
+    call cclutCreateMockTable("sample_table")
+    call cclutAddMockData("sample_table", "sample_table_text")
+
+    call cclutExecuteProgramWithMocks("ut_cclut_table_column_name")
+
+    call cclutAssertvcEqual(CURREF, "test_cclutExecuteProgramWithMocks_same_table_column_name 001",
+        cclut_testRecord->value, "sample_table_text")
+end ;test_cclutExecuteProgramWithMocks_same_table_column_name
+
+/* test_cclutExecuteProgramWithMocks_v9 *********************************************************************
+*  Scenario: Replaces all names (tables and columns) when the version is less than 90300 and revision 9.    *
+************************************************************************************************************/
+subroutine test_cclutExecuteProgramWithMocks_v9(null)
+    record cclut_testRecord (
+        1 table_name = vc
+        1 value = vc
+    ) with protect
+    declare errorMessage = vc with protect, noconstant("")
+    declare errorCode = i4 with protect, noconstant(0)
+    call cclutExecuteProgramWithMocks("ut_cclut_execute_mocks_v9")
+
+    set errorCode = error(errorMessage, 0)
+
+    call cclutAssertVcOperator(CURREF, "test_cclutExecuteProgramWithMocks_v9 001", trim(errorMessage, 3), "regexplike",
+        concat("%CCL-E-26-UT_CCLUT_TABLE_COLUMN_NAME\([^)]+\)[0-9]+:[0-9]+\{ST.", cclut_testRecord->table_name,
+            "}The attribute \(", cclut_testRecord->table_name, "\) does not exist in the table \(ST\)."))
+end ;test_cclutExecuteProgramWithMocks_v9
+
+/* test_cclutExecuteProgramWithMocks_v8_new *****************************************************************
+*  Scenario: Replaces only table names when the version is greater than 81500 and revision 8.               *
+************************************************************************************************************/
+subroutine test_cclutExecuteProgramWithMocks_v8_new(null)
```

Comment:
"There would not need to be four additional support scripts."

Can you provide more elaboration on this point?  It wasn't _just_ because of the cclut prefix that I created those scripts; it was also so that I could test cclutExecuteProgramWithMocks.  Since we're testing framework subroutines, those subroutines are already loaded at the point that this code is called.  To leverage the namespace would mean that there would need to be a program calling cclutExecuteProgramWithMocks.

To use the example you provided, "ut_cclut_table_column_name" would simulate the script-under-test for a regular consumer.  In my case, the _actual_ script-under-test is ut_cclut_execute_mocks_v9 which calls cclutExecuteProgramWithMocks for ut_cclut_table_column_name (to test the function that executes a script-under-test).  So even with the namespace option, I don't see how it sidesteps the need for support scripts.  Given that, I opted to leave the namespace as "cclut" so that it was consistent with the other internal subroutines across the framework.

Therefore,
1. Do you still want me to use the "public" namespace for them instead of "cclut"?
2. Is there some other way you're wanting me to structure these tests that don't involve the support scripts?

Reply:
The ut_cclut_execute_mocks_v* programs do actually not emulate real SUTs.  They include cclutmock.inc.  and invoke cclutExeccutProgramWithMocks directly.  Real SUTs do not do that.

The actual script being tested is ut_cclut_table_column_name as evidenced by the error messages in the assert statements of the unit tests. 

The high-level goal is simply to execute ut_cclut_table_column_name via cclutExeccutProgramWithMocks while controlling the behavior of  cclutGetCurrentVersion and cclutGetCurrentRevision.  

I agree the test I outlined does not achieve that because the namespace needs to be in place before cclutExeccutProgramWithMocks is called and the outline only applies it to the execution of ut_cclut_table_column_name. 

One could modify the outline to set the namespace before calling cclutExecuteProgramWithMocks and to clear it when done, but the right way to do this is to create an actual CCL Unit ut_X for ut_cclut_table_column_name and put it into src/test/resources. Then create a unit test in src/test/ccl that executes cclut_execute_test_case_file on ut_X using **with curnamespace** to control the behaviors for cclutGetCurrentVersion and cclutGetCurrentRevision.  With this approach you could even have ut_X pass some other namespace into its cclutExecuteProgramWithMocks call.